expo/router
# feat: support any URL prefix

# Motivation

- fix #282
- Allow any variation of Expo Go hostname to support deep linking: `127.0.0.1`, `localhost`, `<IP Address>`



# Execution

- Fork `NavigationContainer -> useLinking -> extractPathFromURL` on native.
- Add a custom URL parser which can handle the non-standard URLs used in Expo Go when running in Expo Go.
- Add a custom URL parser on native which doesn't require known prefixes for parsing the pathname.
- Delete manual prefix guessing code.



# Test Plan

- Nothing should change on web platforms.
- Open links `npx uri-scheme open --ios exp://localhost:19006/--/foobar` -> `/foobar`
- Open links `npx uri-scheme open --android exp://127.0.0.1:19006/` -> `/`
